lte/gateway/c/core/oai/tasks/s1ap/s1ap_state_manager.cpp

```diff
@@ -127,14 +123,20 @@ void free_s1ap_state(s1ap_state_t* state_cache_p) {
     OAILOG_ERROR(
         LOG_S1AP, "An error occurred while destroying assoc_id hash table");
   }
-  free_wrapper(reinterpret_cast<void**>(&state_cache_p));
+  free(state_cache_p);
 }
 
 void S1apStateManager::free_state() {
   AssertFatal(
       is_initialized,
       "S1apStateManager init() function should be called to initialize state.");
+
+  if (state_cache_p == nullptr) {
+    return;
+  }
```

Comment:
better to leave this check inside `free_s1ap_state`?

Reply:
Do you think we should have it in both places? Or should an assert be added in `free_s1ap_state` if null is passed in?

Reply:
i think moving the check here and adding the assert on `free_s1ap_state` makes sense, and leaves `free_s1ap_state` with the contract of releasing the task state

Reply:
sounds good...